src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java

```diff
@@ -0,0 +1,200 @@
+
+import static uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.handler.MessageProcessingResultType.POTENTIALLY_RECOVERABLE_FAILURE;
+import static uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.handler.MessageProcessingResultType.SUCCESS;
+import static uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.handler.MessageProcessingResultType.UNRECOVERABLE_FAILURE;
+
+
+@Service
+@Profile("!nosb") // do not register for the nosb (test) profile
+public class PaymentMessageProcessor {
+
+    private static final Logger log = LoggerFactory.getLogger(PaymentMessageProcessor.class);
+
+    private final PaymentMessageHandler paymentMessageHandler;
+    private final IMessageReceiver messageReceiver;
+    private final PaymentMessageParser paymentMessageParser;
+    private final int maxDeliveryCount;
+
+    public PaymentMessageProcessor(
+        PaymentMessageHandler paymentMessageHandler,
+        IMessageReceiver messageReceiver,
+        PaymentMessageParser paymentMessageParser,
+        @Value("${azure.servicebus.payments.max-delivery-count}") int maxDeliveryCount
+    ) {
+        this.paymentMessageHandler = paymentMessageHandler;
+        this.messageReceiver = messageReceiver;
+        this.paymentMessageParser = paymentMessageParser;
+        this.maxDeliveryCount = maxDeliveryCount;
+    }
+
+    /**
+     * Reads and processes next message from the queue.
+     *
+     * @return false if there was no message to process. Otherwise true.
+     */
+    public boolean processNextMessage() throws ServiceBusException, InterruptedException {
+        IMessage message = messageReceiver.receive();
+        if (message != null) {
+            MessageProcessingResult result = process(message);
+            tryFinaliseProcessedMessage(message, result);
+        }
+
+        return message != null;
+    }
+
+    private MessageProcessingResult process(IMessage message) {
+        log.info("Started processing message with ID {}", message.getMessageId());
+
+        PaymentMessage payment = null;
+
+        try {
+            payment = paymentMessageParser.parse(message.getMessageBody());
+            logMessageParsed(message, payment);
+            paymentMessageHandler.handlePaymentMessage(payment);
+            log.info("Processed message with ID {}. Envelope ID: {}", message.getMessageId(), payment.envelopeId);
+            return new MessageProcessingResult(SUCCESS);
+        } catch (InvalidMessageException ex) {
+            log.error("Rejected message with ID {}, because it's invalid", message.getMessageId(), ex);
+            return new MessageProcessingResult(UNRECOVERABLE_FAILURE, ex);
+        } catch (PayHubClientException ex) {
+            logMessageProcessingError(message, payment, ex);
+            MessageProcessingResultType messageProcessingResultType =
+                (ex.getStatus() == HttpStatus.BAD_REQUEST.CONFLICT) ? SUCCESS : POTENTIALLY_RECOVERABLE_FAILURE;
```

Comment:
huh? `HttpStatus.CONFLICT` should be enough shouldn't it? And then `BAD_REQUEST` - unrecoverable. the rest - recoverable¿

Reply:
\+ if it is success, perhaps `logMessageProcessingError` is not quite right? as it is success `log.error` wouldn't really be relevant